### DISCUSSION TITLE
Add the ability to change the position of table columns

### DISCUSSION THREAD

**Discussion by user72356** [supabase/supabase#10268] [Feature Requests]
As my applications grow, I add new columns to tables and those are placed at the end of the list of columns. To ease development I like to keep related columns grouped.

Today we can visually reorder the columns via drag & drop in the Supabase UI and this order is saved within the browser's local storage. The issues with this approach are:

- The custom order is gone if for whatever reason local storage is erased. 
- It isn't reflected on other machines such as my own or my coworkers.
- It isn't reflected in the types generated via the Supabase CLI (https://supabase.com/docs/guides/api/generating-types).
- It isn't reflected in automated reports, which need manual mapping via views or other means.

PostgreSQL itself doesn't have a mechanism by which to change the order of columns. Manually recreating tables is the only way to do this, which isn't practical or advisable for established projects. See https://wiki.postgresql.org/wiki/Alter_column_position

I'm wondering if Supabase could offer this ordering capability through persisting the custom order within the database itself, and then having it reflected to the various clients connecting to the DB (Supabase web UI, Supabase CLI, maybe even third-parties like pgAdmin if you manage to integrate this low enough in the API).

**Comment by saltcod** [comment]
Hi @user72356 ! Thanks for this detailed report. We've been tracking this exact issue internally for a while. We're hoping to get to it soon! I'm going to convert this issue into a discussion in the meantime. Appreciate the feedback!

**Comment by gisborne** [comment]
Rename the table and create a view with the old name with the columns in the order you want.

Postgres should in pretty much every way treat that like the OG table, including you can insert, update, delete, etc.

**Comment by Suman2023** [comment]
hi @saltcod is there any update on this? the answer mention by @gisborne is possible but the draggable would be a better choice.
